modules/phase_field/include/materials/OPInterfaceBarrierMaterial.h

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef OPINTERFACEBARRIERMATERIAL_H
+#define OPINTERFACEBARRIERMATERIAL_H
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class OPInterfaceBarrierMaterial;
```

Comment:
I would like to come up with a better name for this. We already have two materials providing "order parameter interface" barrier contributions. This material is to suppress third phases. How about something like `ThirdPhaseSuppressionMaterial`?